server/utils/cloudFunctions/functions/test/algoliaTests/algoliaFunctions.test.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Description: Unit tests for algoliaFunctions.js
+ *              Uses stubbing to mock algolia save/delete.
+ *              Uses sinon/chai to spy on calls and validate behavior.
+ *              compile with `npm run test`.
+ * Author: Robert Marcus
+ * Date: July 11, 2020
+ */
+
+// Import the testing environment configuration
+const { assert, sinon, firestoreWrap, firestoreMock } = 
+    require('../testConfig');
+
+// Import the generic cloud functions.
+const devAlgoliaFunctions = 
+    require('../../algoliaCloudFunctions/devAlgoliaFunctions');
+const { DEV_ADS_INDEX } = 
+    require('../../algoliaCloudFunctions/algoliaConfig');
+
+// Import mock saveObject, mock deleteObject, and helpers.
+const { mockDeleteObject, mockSaveObject } = require('./algoliaMocks');
+
+// Restore the stubs after each test. 
+afterEach(() => {
+  sinon.restore();
+});
+
+describe("test_devCreateRecord", () => {
```

Comment:
I think you can remove "dev" from these test descriptions. So just test_CreateRecord for example.
It is expected that unit tests run on the dev not prod.